src/components/05_pages/NodeForm/index.js

```diff
@@ -100,7 +103,17 @@ class NodeForm extends React.Component {
   };
 
   onSave = () => {
-    this.props.onSave(this.state.entity.data);
+    const missingFields = this.resolveMissingRequiredFields();
+    if (missingFields.length) {
+      this.props.setMessage(
+        missingFields.length > 1
+          ? `The following fields are missing, ${missingFields.join(', ')}.`
```

Comment:
Nice!

Reply:
This felt easier that trying to pluralize some bits of the string.

Reply:
Yeah, never ever try to split up text to some bits of the string. This makes it much harder later to introduce localization of the UI.